components/automate-ui/src/app/entities/environments/environment.effects.ts

```diff
@@ -98,4 +103,30 @@ export class EnvironmentEffects {
         message: `Could not delete environment: ${msg || error}`
       });
     }));
+
+  @Effect()
+  updateEnvironment$ = this.actions$.pipe(
+    ofType(EnvironmentActionTypes.UPDATE),
+    mergeMap(({ payload }: UpdateEnvironment) =>
+      this.requests.updateEnvironment(payload).pipe(
+        map((resp: Environment) => new UpdateEnvironmentSuccess(resp)),
+        catchError((error: HttpErrorResponse) =>
+          observableOf(new UpdateEnvironmentFailure(error))))));
+
+  @Effect()
+  updateEnvironmentSuccess$ = this.actions$.pipe(
+    ofType(EnvironmentActionTypes.UPDATE_SUCCESS),
+    map(({ payload: environment }: UpdateEnvironmentSuccess) => new CreateNotification({
+      type: Type.info,
+      message: `Updated environment ${environment.name}.`
+    })));
+
+  @Effect()
+  updateEnvironmentFailure$ = this.actions$.pipe(
+    ofType(EnvironmentActionTypes.UPDATE_FAILURE),
+    filter(({ payload }: UpdateEnvironmentFailure) => payload.status !== HttpStatus.CONFLICT),
+    map(({ payload }: UpdateEnvironmentFailure) => new CreateNotification({
+      type: Type.error,
+      message: `Could not Update notification: ${payload.error.error || payload}.`
```

Comment:
change this error message, relace with the environment.

Reply:
updated the changes